lib/tapioca/compilers/requires_compiler.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+# typed: true
+
+require_relative '../sorbet_config_parser'
+
+module Tapioca
+  module Compilers
+    class RequiresCompiler
+      extend T::Sig
+
+      sig { params(sorbet_path: String).void }
+      def initialize(sorbet_path)
+        @sorbet_path = sorbet_path
+      end
+
+      sig { returns(String) }
+      def compile
+        config = SorbetConfig.parse_file(@sorbet_path)
+        files = collect_files(config)
+        files.flat_map do |file|
+          collect_requires(file).reject do |req|
+            name = collect_required_name(T.cast(req, String))
+            next unless name
+            name_in_project?(files, T.must(name[1]))
+          end
+        end.sort.uniq.join
+      end
+
+      private
+
+      sig { params(config: SorbetConfig).returns(T::Array[String]) }
+      def collect_files(config)
+        config.paths.flat_map do |path|
+          path = (Pathname.new(@sorbet_path) / "../.." / path).cleanpath
+          if path.directory?
+            Dir.glob("#{path}/**/*.rb", File::FNM_EXTGLOB).reject do |file|
+              file_ignored_by_sorbet?(config, file)
+            end
+          else
+            [path.to_s]
+          end
+        end.sort.uniq
+      end
+
+      sig { params(file_path: String).returns(T::Enumerable[String]) }
+      def collect_requires(file_path)
+        File.read(file_path).lines.select do |line|
```

Comment:
I wonder if it might make more sense to have a single regex and to `map` all lines to a (potential) require coming from that line. We can then compact to get rid of all the `nil`s. That would simplify the code by getting rid of `collect_required_name` and removing the `next unless name` in the loop using this.

That would also allow us to extract the regex to a constant so that it is more prominent what we are matching.

Also, it would allow us to generate a require file that is more uniform since would be in control of formatting the `require "..."` lines.

Reply:
I went with formatting at first, but not doing it was actually a better way to show the inconsistencies in the client coding style.

Reply:
Since we now collect requires with possible indentation so I also needed to remove the indent. Keeping the original `"` or `'` was too much trouble so I went with the formatting after all.